engine/src/main/java/org/camunda/bpm/engine/impl/dmn/entity/repository/DecisionRequirementsDefinitionQueryImpl.java

```diff
@@ -193,18 +194,24 @@ public DecisionRequirementsDefinitionQuery orderByTenantId() {
 
   @Override
   public long executeCount(CommandContext commandContext) {
-    checkQueryOk();
-    return commandContext
-      .getDecisionRequirementsDefinitionManager()
-      .findDecisionRequirementsDefinitionCountByQueryCriteria(this);
+    if (DecisionDefinitionQueryImpl.isDmnEnabled()) {
```

Comment:
:x: The code will be more clean if you use directly the `commandContext`. An alternative will be extract the the dmn check in a util class.
```suggestion
    if (commandContext.getProcessEngineConfiguration().isDmnEnabled()) {
```